app/helpers/sessions_helper.rb

```diff
@@ -4,15 +4,36 @@ def log_in user
   end
 
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by id: user_id
+    elsif user_id = cookies.signed[:user_id]
+      user = User.find_by id: user_id
+      if user&.authenticated? cookies[:remember_token]
+        log_in user
+        @current_user = user
+      end
+    end
   end
 
   def logged_in?
     current_user.present?
   end
 
+  def forget user
+    @current_user && user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
   def log_out
+    forget current_user
     session.delete :user_id
     @current_user = nil
   end
+
+  def remember user
+    @current_user && user.remember
```

Comment:
method remember này thực thi sau khi xử lý login user -> lúc này current_user đã có giá trị, vậy chỗ này có cần thiết phải check lại current_user nữa ko em?

Reply:
Dạ tại ko dùng cái nó báo undefined method `remember' for nil:NilClass á chị

Reply:
vậy thì phải check user truyền từ param vào method em nha.
mà ở

Reply:
Da dc roi chi